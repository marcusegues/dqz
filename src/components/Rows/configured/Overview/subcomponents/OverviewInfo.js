// @flow
import React from 'react';
// $FlowFixMe
import { View } from 'react-native';
import { CardRowText } from '../../../../QuestionAnswer/cards/subcomponents/CardRowText';
import { CardRowSubText } from '../../../../QuestionAnswer/cards/subcomponents/CardRowSubText';

type OverviewInfoProps = {
  title: string,
  subtitle: string,
  children: any,
};

export const OverviewInfo = ({
  title,
  subtitle,
  children,
}: OverviewInfoProps) => {
  return (
    <View
      style={{
        flex: 1,
        flexDirection: 'column',
        alignItems: 'flex-start',
        paddingRight: 5,
      }}
    >
      <CardRowText text={title} />
      <CardRowSubText text={subtitle} />
      {children}
    </View>
  );
};
