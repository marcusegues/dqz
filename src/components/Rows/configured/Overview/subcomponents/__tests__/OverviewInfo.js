// @flow
import React from 'react';
// $FlowFixMe
import { Text } from 'react-native';
import renderer from 'react-test-renderer';
import { OverviewInfo } from '../OverviewInfo';

describe('OverviewInfo', () => {
  test('OverviewInfo renders correctly with children', () => {
    const tree = renderer
      .create(
        <OverviewInfo title="test title" subtitle="test subtitle">
          <Text>Children</Text>
        </OverviewInfo>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
