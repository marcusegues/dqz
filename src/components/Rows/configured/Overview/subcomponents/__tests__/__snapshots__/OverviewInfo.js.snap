// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OverviewInfo OverviewInfo renders correctly with children 1`] = `
<View
  style={
    Object {
      "alignItems": "flex-start",
      "flex": 1,
      "flexDirection": "column",
      "paddingRight": 5,
    }
  }
>
  <View
    style={Object {}}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#24253D",
          "fontFamily": "roboto_medium",
          "fontSize": 22,
        }
      }
    >
      test title
    </Text>
  </View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#898989",
        "fontFamily": "roboto_regular",
        "fontSize": 18.857142857142858,
      }
    }
  >
    test subtitle
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    Children
  </Text>
</View>
`;
